### PR TITLE
made epoch start metrics persistent and/or fetch from epoch bootstrapper

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
Made metrics related to nonce and round at epoch start persistent

### CODE REVIEW

File: epochStart/bootstrap/process.go [hunk 1]
```diff
@@ -878,6 +881,11 @@ func (e *epochStartBootstrap) createRequestHandler() error {
 	return err
 }
 
+func (e *epochStartBootstrap) setEpochStartMetrics() {
+	e.statusHandler.SetUInt64Value(core.MetricNonceAtEpochStart, e.epochStartMeta.Nonce)
```
Comment (iulianpascalau) [inline epochStart/bootstrap/process.go:885]: shouldn't we also rewrite the metrics for `erd_nonces_passed_in_current_epoch` and `erd_rounds_passed_in_current_epoch` ?

Comment (bogdan-rosianu) [reply epochStart/bootstrap/process.go:885]: No, those are calculated when calling the API endpoint. So it is ok

File: epochStart/bootstrap/common.go [hunk 2]
```diff
@@ -88,6 +89,9 @@ func checkArguments(args ArgsEpochStartBootstrap) error {
 	if args.GeneralConfig.EpochStartConfig.MinNumConnectedPeersToStart < minNumConnectedPeers {
 		return fmt.Errorf("%s: %w", baseErrorMessage, epochStart.ErrNotEnoughNumConnectedPeers)
 	}
+	if check.IfNil(args.StatusHandler) {
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply epochStart/bootstrap/common.go:92]: ok, done

File: cmd/node/main.go [hunk 3]
```diff
@@ -863,6 +863,7 @@ func startNode(ctx *cli.Context, log logger.Logger, version string) error {
 		Rounder:                    rounder,
 		AddressPubkeyConverter:     addressPubkeyConverter,
 		LatestStorageDataProvider:  latestStorageDataProvider,
+		StatusHandler:              coreComponents.StatusHandler,
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply cmd/node/main.go:866]: it is ok. on main.go, line 738 we see a :
`coreComponents.StatusHandler = statusHandlersInfo.StatusHandler`
and the bootstrapper args are written on line 839

[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply cmd/node/main.go:866]: the `nil` implementation of status handler is used when returning the Core Components but the value is filled after status handlers are created

File: epochStart/bootstrap/common.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply epochStart/bootstrap/common.go:92]: also added nil checks when using the status handler inside bootstrapper, as it is only an additional component.

File: cmd/node/main.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply cmd/node/main.go:866]: ok

File: epochStart/bootstrap/process.go (hunk 1 above)
Comment (iulianpascalau) [reply epochStart/bootstrap/process.go]: ok

File: epochStart/bootstrap/process.go [hunk 4]
```diff
@@ -891,8 +892,10 @@ func (e *epochStartBootstrap) createRequestHandler() error {
 }
 
 func (e *epochStartBootstrap) setEpochStartMetrics() {
-	e.statusHandler.SetUInt64Value(core.MetricNonceAtEpochStart, e.epochStartMeta.Nonce)
-	e.statusHandler.SetUInt64Value(core.MetricRoundAtEpochStart, e.epochStartMeta.Round)
+	if !check.IfNil(e.statusHandler) && e.epochStartMeta != nil {
```
Comment (iulianpascalau) [inline epochStart/bootstrap/process.go:895]: do we still need the check !check.IfNil(e.statusHandler) ?

**Pull request merged**